### ISSUE TITLE
Add support for ChartJS

### ISSUE THREAD

**Issue by contactashish13** [Codeinwp/visualizer#373]
https://wordpress.org/plugins/wp-charts/ uses https://www.chartjs.org/ but the support on https://wordpress.org/support/plugin/wp-charts/ is poor

**Comment by ineagu** [comment]
the plugin was just sold, so might pick steam. A reason to add support is that those seem to be much cleaner/faster, no jquery vs google charts.

**Comment by contactashish13** [comment]
PRs https://github.com/Codeinwp/visualizer/pull/425 and https://github.com/Codeinwp/visualizer-pro/pull/168

There is a slight change in the workflow to support this. The option to choose the library (default would be Google Charts for all except DataTable for table charts and ChartJS for polar and radar charts) can be found in the chart type selection screen. Selecting a chart will automatically disable/enable the libraries supported.

![image](https://user-images.githubusercontent.com/12953439/59329455-8874f100-8d0c-11e9-9eac-f8a8f108f60a.png)

@mileamadalin since this is a big change and can potentially break sites, I'll need some help to test this. Please see how this can be done.

**Comment by mileamadalin** [comment]
I will check it next week and come back.

**Comment by SilviuDinu** [comment]
I have tested how creating / displaying a chart created with ChartJS works and I have made a few videos with the results:

- https://embed.vidyard.com/share/d1k1mHcVoUapgpf2CL2hFg

- https://embed.vidyard.com/share/tyyvLMgZQLeyQ3kFhPUe2Y?

- https://embed.vidyard.com/share/d1k1mHcVoUapgpf2CL2hFg

The charts seem to behave strange when editing them. Hovering over certain areas make the chart black and weird for a split second. This does not happen all the time, so I'm not sure what is causing this. The testing environment is new, and I'm using the default WordPress theme.

When displaying them they look okay, except when resizing the browser window. The text and title is scaling correctly, but the chart itself remains at the same size (not much of a problem, but not as expected).

**Comment by SilviuDinu** [comment]
Update: It seems there are still some hovering issues, but they behave slightly different (the background does not turn black anymore):

- https://embed.vidyard.com/share/owhUzv4p4Ke382X8wMdBaM? - Creating and editing the bar chart. Issue when hovering.

- https://embed.vidyard.com/share/m7CrgbW8jLRT4fyCSDzVNf? - Issue with the Pie chart on hover. Also, choosing the Donut option also adds up to the hover issue. It's like there are multiple charts existing at the same time one on top of the other.

- http://prntscr.com/o9yjsb - the pie chart seems to behave well when displaying it on a post using the shortcode block, but it does seem a bit too large from my perspective as an user. 

Personal thought -> the issue with the hover effect only occurs if you edit the chart. If you save it as it is and press on Edit again, it behaves normally, but if you change another thing about it, the hovering issue occurs again.

The zooming factor thing has not been addressed yet so it still exists.

**Comment by SilviuDinu** [comment]
I pulled the latest changes and I'm not noticing the hovering issues from above anymore:

For the first 2 videos, I used an already existing instance and the ShopIsle theme (Lite). The third video was made on an instance where I had the previous version of the plugin, where the hover issue used to happen. For the last video, I used a fresh instance with the default WordPress theme

- https://embed.vidyard.com/share/H375tZW7bv9gx3MtpXuTS2? - Updating to the latest version and displayed a Pie chart using the shortcode method. The chart behaves correctly and the chart now adjusts to the zoom factor of the browser window. However, i cannot use the Visualizer block to "display existing charts" cc @HardeepAsrani 

- https://embed.vidyard.com/share/k6x1AFcJVYLPTTxcUpKjFp? - I noticed that sometimes when I play with the browser window zooming factor, the charts get resized accordingly, but when i zoom back to the default 100% value, the charts are blurred (coming from 25% to 100%).

- https://embed.vidyard.com/share/biNYYG4aQvKTbbiX1R5vpy? - on an old instance, with previousely created charts, I created, configured and edited a new pie chart (with the latest version). the theme here is the default WordPress theme.

- https://embed.vidyard.com/share/oBHk9jUtxfAb7XjH3A1WWY?  - new instance. Everything seems to run as expected.

For the old instance case, the website did not seem to break.
